spi/storage/storage.go

```diff
@@ -33,13 +33,40 @@ type StoreConfiguration struct {
 	TagNames []string `json:"tagNames,omitempty"`
 }
 
+// SortOrder specifies the sort order of query results.
+type SortOrder int
+
+const (
+	// SortAscending indicates that the query results must be sorted in ascending order.
+	SortAscending SortOrder = iota
+	// SortDescending indicates that the query results must be sorted in descending order.
+	SortDescending
+)
+
+// SortOptions sets the order that results from an Iterator will be returned in. Sorting is based on the tag values
+// associated with the TagName chosen below. The TagName you use below can be the same as the one you're querying on,
+// or it can be a different one. However, you must ensure that the TagName set below is in the
+// Store's StoreConfiguration before trying to use it for sorting, or unexpected behaviour may occur.
+// If tag value strings are decimal numbers, then the sorting will be based on their numerical value instead of
```

Comment:
(so store implementations need to make sure that numbers aren't sorted like strings, which results in a different order than based on their numerical value)